### PR TITLE
Add scoring logic

### LINKED ISSUE #470: Create SR2020 game score format
The format for the game data YAML files which record match outcomes needs to be designed.

### Original

[comp/game/score-format](https://github.com/srobo/recurring-tasks/blob/master/comp/game/score-format.yaml)

### LINKED ISSUE #471: Write scoring scripts for the SR2020 game
The scoring script provides automatic computation of the scores based on the data input from the score sheets.
See https://github.com/PeterJCLaw/srcomp/wiki/Competition-Preparation#implement-scorepy for details.

### Original

[comp/game/score-script](https://github.com/srobo/recurring-tasks/blob/master/comp/game/score-script.yaml)

### Dependencies

 * #470 Create SR2020 game score format

### PR DESCRIPTION
**Pull request by PeterJCLaw**
Fixes https://github.com/srobo/tasks/issues/471.
Fixes https://github.com/srobo/tasks/issues/470.

### CODE REVIEW

File: scoring/score.py
Comment (RealOrangeOne) [inline scoring/score.py]: I don't think this is quite correct.

> If tokens of both colours are present in a scoring zone,  the value of each token in that scoring zone is reduced from 3 to 1.

This looks to half the points at the end, making the tokens worth 1.5 points rather than 1 point.

eg https://github.com/srobo/sr2020-comp/pull/1/files#diff-e58af9df2705fcda0714a31abe6a7985R141 should be 2 points rather than 3.

Comment (PeterJCLaw) [reply scoring/score.py]: Ah, good spot. This is a failing in my recollection of the rules. I even had the rulebook open when I was doing this! :man_facepalming:

Comment (PeterJCLaw) [reply scoring/score.py]: c44f3dc

**Review by RealOrangeOne — APPROVED** [review]
This looks good to me! Although worth getting a second (third) pair of eyes on it too, just so we're sure!

**Comment by PeterJCLaw** [comment]
@sedders123 would you have some time to have a look at this? Would be great to get another set of eyes over it before we merge it.

**Comment by PeterJCLaw** [comment]
Ah, looks like this doesn't include the bonus point for leaving the scoring zone.

**Comment by PeterJCLaw** [comment]
> Ah, looks like this doesn't include the bonus point for leaving the scoring zone.

This is fixed in a5f5320.

**Pull request merged**
